File: app/src/main/java/team58/cs2340/donationtracker/models/LocationType.java
package team58.cs2340.donationtracker.models;

/**
 * This class represents various location types
 */
public enum LocationType {
    DROPOFF("Drop Off"),
    STORE("Store"),
    WAREHOUSE("Warehouse");

    private final String typeName;

    LocationType (String typeName) {
        this.typeName = typeName;
    }

    /**
     * Function to return type name
     * @return type name to be returned
     */
    public String getTypeName() {
        return this.typeName;
    }

    /**
     * Function to return location type enum from string
     * @param text string to be converted
     * @return location type enum of string
     */
    public static LocationType fromString(String text) {
        for (LocationType l : LocationType.values()) {
            if (l.typeName.equalsIgnoreCase(text)) {
                return l;
            }
        }
        return null;
    }
}

File: app/src/main/java/team58/cs2340/donationtracker/models/LocationType.kt
package team58.cs2340.donationtracker.models

/**
 * This class represents various location types
 */
enum class LocationType private constructor(private val typeName: String) {
    DROPOFF("Drop Off"),
    STORE("Store"),
    WAREHOUSE("Warehouse");


    companion object {

        /**
         * Function to return location type enum from string
         * @param text string to be converted
         * @return location type enum of string
         */
        fun fromString(text: String): LocationType? {
            for (l in LocationType.values()) {
                if (l.typeName.equals(text, ignoreCase = true)) {
                    return l
                }
            }
            return null
        }
    }
}
